DragaliaAPI/Services/Game/SummonService.cs

```diff
@@ -94,48 +95,144 @@ float pity
+    public List<AtgenRedoableSummonResultUnitList> GenerateSummonResult(
+        int numSummons,
+        int bannerId
+    )
     {
         return this.GenerateSummonResult(
             numSummons,
             10,
             0.0f /*,new(new(), new(), 6.0d, 80.0d) */
+            ,
+            bannerId
         );
     }
 
+    public Dictionary<int, Tuple<int, int>> GetSummonData(int bannerId)
+    {
+        Dictionary<int, Tuple<int, int>> selectedUnitDict = new Dictionary<int, Tuple<int, int>>();
+        var bannerSummonData = MasterAsset.SummonData.Get(bannerId);
+
+        if (bannerId == 1020203)
+        {
+            // If banner-ID matches the standard-banner, choose from a pool of every unit.
+            selectedUnitDict = Enum.GetValues(typeof(Charas))
+                .Cast<Charas>()
+                .Where(chara => (int)chara != 0)
+                .ToDictionary(chara => (int)chara, chara => Tuple.Create(1, 5)); // Every unit is a chara.
+        }
+        else
+        {
+            void AddToDict(int unitId, int unitType)
+            {
+                if (
+                    unitId != 0
+                    && (unitType == 1 || unitType == 2)
+                    && !selectedUnitDict.ContainsKey(unitId)
+                )
+                {
+                    selectedUnitDict[unitId] = Tuple.Create(unitType, 5);
+                }
+            }
+
+            AddToDict(bannerSummonData.PickupUnitId1, bannerSummonData.PickupUnitType1);
+            AddToDict(bannerSummonData.PickupUnitId2, bannerSummonData.PickupUnitType2);
+            AddToDict(bannerSummonData.PickupUnitId3, bannerSummonData.PickupUnitType3);
+            AddToDict(bannerSummonData.PickupUnitId4, bannerSummonData.PickupUnitType4);
+        }
+
+        // Add also charas with rarity 3 or 4
+        foreach (var chara in MasterAsset.CharaData.Enumerable)
+        {
+            if (chara.Rarity == 3 || chara.Rarity == 4)
+            {
+                selectedUnitDict[(int)chara.Id] = Tuple.Create(1, chara.Rarity);
+            }
+        }
+
+        // Choose dragon. Pool of dragons isn't limited until we have a standard banner pool.
+
+        foreach (var dragon in MasterAsset.DragonData.Enumerable)
+        {
+            string dragonIdString = dragon.Id.ToString();
+
+            if (
+                dragon.Rarity == 3
+                || dragon.Rarity == 4
+                || dragon.Rarity == 5 && dragon.IsPlayable == true
+            )
+            {
+                selectedUnitDict[(int)dragon.Id] = Tuple.Create(2, dragon.Rarity);
+            }
+        }
+
+        return selectedUnitDict;
+    }
+
     public List<AtgenRedoableSummonResultUnitList> GenerateSummonResult(
         int numSummons,
         int summonsUntilNextPity,
-        float pity /*,
-        BannerSummonInfo bannerInfo */
+        float pity,
+        int bannerId
     )
     {
         List<AtgenRedoableSummonResultUnitList> resultList = new();
```

Comment:
The game guarantees you get a 4* if you do a tenfold (but not with single tickets) -- probably need to ensure this happens here, maybe a `forceFourStar` flag?